### PR TITLE
[stackalloc-init] Improve error tolerance when parsing invalid implicit stackalloc arrays

### PR DESCRIPTION
**Pull request by alrz**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
> The error message should probably be changed. I'll wait for review feedback before changing it.

Agreed.

**Comment by sharwell** [comment]
@jinujoseph Shouldn't this be assigned to the compiler team instead of me?

**Comment by jcouv** [comment]
Tagging @VSadov @gafter for review. Thanks

**Comment by VSadov** [comment]
LGTM, but we need a better error. Any good ideas?


Example of where the current error might sound strange: 
(there is no type here)

```
                // (4,28): error CS1575: A stackalloc expression requires [] after type
                //         var x = stackalloc[3,] { 1, 2, 3 };
```

**Comment by jcouv** [comment]
I think the error message we use elsewhere is something like "expected ]". Maybe that's better?

**Comment by VSadov** [comment]
@jcouv  - just to clarify, are you suggesting simply putting an error on the first unexpected token and produce some standard  " expected ']' "  ?

Considering that the stackalloc is a fairly advanced feature, this would seem sufficient.

**Comment by jcouv** [comment]
Yes, that's what I was suggesting.

For context, here's the error we just added for implicit array creation (https://github.com/dotnet/roslyn/pull/24706/files#diff-0d8acdaeed6fd4e02198f6e3745d6d2dR286): "Invalid rank specifier: expected ',' or ']'".

For stackalloc (`var x = stackalloc[3,] { 1, 2, 3 };`), we could use "Invalid rank specificier: expected ']'".

**Comment by alrz** [comment]
>Invalid rank specificier: expected ']'".

to be clear, should this be a new error or we should update the existing one?

**Comment by jcouv** [comment]
@alrz It seems that `error CS1575: A stackalloc expression requires [] after type` is still useful. So this would be a new error code and message.

**Comment by gafter** [comment]
@VSadov There are two reviews of this PR now.

**Comment by alrz** [comment]
I should add the new error. will push it asap.

**Comment by gafter** [comment]
Still looks good to me.

File: src/Compilers/CSharp/Portable/CSharpResources.resx [hunk 1]
```diff
@@ -5256,4 +5256,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureStackAllocInitializer" xml:space="preserve">
     <value>stackalloc initializer</value>
   </data>
+  <data name="ERR_InvalidStackAllocArray" xml:space="preserve">
+    <value>"Invalid rank specificier: expected ']'</value>
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/CSharpResources.resx]: >specificier [](start = 25, length = 11)

"specificier"???  I think this is a typo.

Comment (alrz) [reply src/Compilers/CSharp/Portable/CSharpResources.resx]: oops.

**Pull request merged**